### ISSUE TITLE
[InlineEdit]: New TEDI-Ready component

### ISSUE THREAD

**Issue by airikej** [TEDI-Design-System/react#199]
https://github.com/TEHIK-EE/tedi-design-system/issues/193 
[Figma](https://www.figma.com/design/jWiRIXhHRxwVdMSimKX2FF/TEDI-READY-(work-in-progress)?node-id=5881-57032&m=dev)

- no API changes
- Move to TEDI-Ready
- Add unit test
- improve WCAG for toggle button
- improve props descriptions
- add documentation to ZeroHeight

| Name          | Required/Optional | Description                                                                                  | Type                                                                                     | Possible Values                          | Default |
|---------------|-------------------|----------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------|------------------------------------------|---------|
| `className`   | Optional          | Classname for the button/displayed content                                              | `string`                                                                                 | Any valid CSS class name                 |         |
| `content`     | Required          | Content displayed when the field is hidden.                                                  | `React.ReactNode`                                                                        | Any valid React node                     |         |
| `fieldType`   | Required          | Specifies the type of hidden field.                                                          | `'textfield' \| 'select' \| 'datetime'`                                                | `'textfield'`, `'select'`, `'datetime'` |         |
| `fieldOptions`| Required          | Configuration options for the specified field type.                                          | `TextFieldProps` (for `textfield`), `SelectProps` (for `select`), `DateTimePickerProps` (for `datetime`) | Based on the `fieldType`                |         |

**Comment by airikej** [comment]
@karolin-kivisikk @mart-sessman I've put together a mock of this component here:
https://storybook.tedi.ee/react/feat/199-hiddenfield-new-tedi-ready-component/?path=/docs/tedi-ready-components-form-editablefield--docs
a true directive like in Angular isn't really possible in React, so I went with a wrapper component instead. It can turn a range of form components into editable-in-place fields. you can give your thoughts and suggestions on how to take this forward

**Comment by karolin-kivisikk** [comment]
Design review: 
1. Mul kana kitkuda selle nimetusega, ma ei tea endiselt mis see hea nimetus oleks ka aga Editable field on selles osas nii üldine nimetus et kõik fieldid on tegelt editable, ta on nagu Ghost field? Disainerid pakkusid ka nimetusi aga need pole ka liiga head: On hover field, hover editable field, hover input, hover edit style
2. Kas see nüüd on loodud eraldi komponendina või see on css stiil mida saab igale komponendile rakendada? Ma ise näen et see võiks olla eraldi stiil mida ükskõik mis komponendile saab rakendada ehk et eemaldab borderi ja paddingu ja hoverdades tekitab tausta jne ehk et siis saab olemasoleva inputi kergelt selleks muuta. 
3. Figmas on stiilijuhis ka olemas: https://www.figma.com/design/jWiRIXhHRxwVdMSimKX2FF/TEDI-READY-2.70.88?node-id=9938-87560&t=nBwNRsGgWvOkBlKR-4
4. Edit ikooni olemasolu on hea mõte, parem UX, aga võiks saada ka ilma selleta lisada - mõnikord on tarvis ilma. See peaks WCAG kontrolli ka minema kindlasti. 
5. Võiks näitena olla rohkem erinevat tüüpi vormivälja sisusid ka, multiselect, dropdown jms, aga radiod ja checkboxid peaks olema ikkagi lihtsalt dropdownid sellises vaates on veidi veider kui terve list tekib sinna nähtavale (see võimekus võib jääda aga lihtsalt mitte siia näitena).
6. Kellaaja valik on jube pahasti siin, vormis sees ei peaks selline kellaaja valik olema see on siiski poigem dropdowni mõeldud ehk et siin võiks ka avada dropdoni kas lihtsalt kellaaegade valikuga või timepickeriga
7. Muuda edit nupu font size 18px peale, lisaks peaks olema need nupud sinised et läheks kokku tedi nuppude neutral stiiliga
8. Label võiks jääda oma kohapeale isegi kui input läheb edit olekusse, hetkel hüppab:

<img width="731" height="229" alt="Image" src="https://github.com/user-attachments/assets/0f5af6c8-594c-4ebd-8d79-fbbd7df8650b" />
<img width="786" height="370" alt="Image" src="https://github.com/user-attachments/assets/91194530-10b2-4f6d-8738-4f2a109656ff" />

9. Disabled välja puhul peaks tekst antud juhul mitte muutuma, tegu peaks olema täpselt sellise stiiliga nagu text group on ehk kuna tegu pole inputi väljaga siis kasutaja nagunii ei näe välja ennast ning tekst ei peaks olema kuidagi heledam, tekst võib olla tavatekst millel lihtsalt puudub hover ja edit võimekus. 
10. Kuupäeva formaat vale, 22.03.2026 <- kuul peaks 0 ka ees olema

**Comment by airikej** [comment]
https://storybook.tedi.ee/react/feat/199-hiddenfield-new-tedi-ready-component/?path=/docs/tedi-ready-components-form-editable--docs
tegin parandusi, edit ikoon ka välja lülitatav nüüd propiga

nime osas arendajate koosolekul arutasime, et vähemalt reactis oldi pigem selle poolt, et oleks ikkagi komponent, nime osas oldi nõus, et editableFIELD oleks justkui vale, sest see on nn wrapper komponent, mis muudab tema sees vormi komponendi lihtsalt muudetavaks, hetkel panin editable, aga need disainerite pakutud on veel hullemad, need ei meeldinud kellelegi. angularis polnud kellegil mingit arvamust, ju ei ole seal hetkel vajadust sellelele komponendile.
multiselect/selecti nn editable olekut tahaks ka challengida, minu arust ei peaks ikkagi tag-id seal niimoodi nähtaval olema, sest see justkui indikeerib ka interaktiivsust, et justkui closable tagid, aga klikkad peale tekib vorm? mõtleks selle ka disainis veits läbi. dropdown pole ka vormi element - ei käi vormide juurde, seega seda näidet sinna ei pannud

**Comment by karolin-kivisikk** [comment]
1. Hea et lisasid võimekuse edit nupp teksti järele ka lisada, aga seal peaks olema min 8px vahe. 
<img width="347" height="239" alt="Image" src="https://github.com/user-attachments/assets/6f96e72e-c588-4d63-a718-ffae8d25c31c" />

2. Teha mõni näide ilma nähtava edit nuputa ka
3. Figmas sai tagid ära vahetatud read-only tagide vastu
4. Kas on option teha ka selline lahendus kus kasutajale kuvatakse staatust aga kui peale klikkab siis avaneb select kust siis saab valida staatuse?
5. Nimetuse ideid veel: 
- EditablePreview
- ClickToEdit
- InlineEdit - see äkki?
- EditableValue
- ReadModeField
- ClickToEdit
- EditOnClick

Ma jätan selle taski veel enda nimele kuna disainis võib see nädal tulla ka muudatusi kui WCAG testija ütleb et midagi ei sobi, saatsin figmas ka ülevaatamisele.

**Comment by karolin-kivisikk** [comment]
Sain WCAG testimisest infot et "Ilma ikooni kasutamata rikub nõudeid, sest element peaks olema avastatav ilma, et kasutaja nö kogemata satub selle peale."

Ehk et eemaldan Figmast ka võimaluse siis seda edit ikooni välja lülitada siis

**Comment by airikej** [comment]
@karolin-kivisikk @Liberiina https://storybook.tedi.ee/react/feat/199-hiddenfield-new-tedi-ready-component/?path=/docs/tedi-ready-components-form-inlineedit--docs
viisin Figmaga vastavusse näited, dokumentatsiooni lükkasin eraldi faili, et ei risustaks sissejuhatust väga. hideEditIcon propi jätsin alles, kuid lisasin klausli, et seda kasutades ei passi wcag ja kasutamine omal vastutusel

**Comment by karolin-kivisikk** [comment]
1. Kitsal ekraanil storyd üle vaadata:
<img width="470" height="680" alt="Image" src="https://github.com/user-attachments/assets/cf105e76-521e-4ad1-a352-9afc9822d9f7" />

2. States storyd ei vasta hetkel states-ile, st active ja hover ja focus ei tule välja kõigil.

<img width="1295" height="440" alt="Image" src="https://github.com/user-attachments/assets/dad74b8e-99c4-4030-8e61-e60458038faf" />

Muud asjad tunduvad korras.

**Comment by airikej** [comment]
@karolin-kivisikk ei suutnud states probleemi reproda, proovisin erinevaid brausereid, äkki proovi hard refreshi teha korraks
